File: 02-Algoritmos de ordenação/BubbleSort.c
#include <stdio.h>

#define MAX 5

int
main(void) {
    int x[MAX];

    for (int i = 0; i < MAX; ++i) {
        printf("Digite o %d número: ", i+1);
        scanf("%d", &x[i]);
    }


    // Ordenando o vetor em ordem crescente
    for (int i = 0; i <= MAX; ++i) {
        for (int j = 0; j < MAX-1; ++j) {
            if (x[j] > x[j+1]) {
                int aux = x[j];
                x[j] = x[j+1];
                x[j+1] = aux;
            }
        }
    }


    // Exibe o vetor na tela
    for (int i = 0; i < MAX; ++i) {
        printf("%d\t", x[i]);
    }

    return 0;
}

File: 02-Algoritmos de ordenação/BubbleSort.java
import java.util.*;

/**
 * O bubble pega cada valor que está contido no vetor e
 * faz um comparação um a um até todo o vetor está ordeanado
 */

public class BubbleSort {
    public static void main(String[] args) {
        int x[] = new int[5];
        Scanner input = new Scanner(System.in);

        for (int i = 0; i < x.length; ++i) {
            System.out.printf("Digite o '%d' número: ", i+1);
            x[i] = input.nextInt();
        }

        /**
         * Ordenando de forma crescente
         */
        for (int i = 0; i <= x.length; ++i) {
            for (int j = 0; j < x.length-1; j++) {
                if (x[j] > x[j+1]) {
                    int aux = x[j];
                    x[j] = x[j+1];
                    x[j+1] = aux;
                }
            }
        } 

        // Imprime o vetor ordenado na tela
        for (int i = 0; i < x.length; ++i) {
            System.out.printf("%d\t", x[i]);
        }
        
        input.close();
    }
}